core/src/providers/openai.rs

```diff
@@ -1149,11 +1171,12 @@ pub async fn streamed_chat_completion(
                     finish_reason: None,
                 })
                 .collect::<Vec<_>>(),
+            usage,
         };
 
         for i in 0..guard.len() {
             let a = guard[i].clone();
-            if a.choices.len() != f.choices.len() {
+            if a.choices.len() != f.choices.len() && a.usage.is_none() {
```

Comment:
Why this condition?

Reply:
that one is not very clear .. when requesting the token usage, we receive a dedicated fake chunk with only usages, but no completion/choices .. and as f is a clone of the first chunk and contain one choice, this test fails if we use it on that usage chunk. another solution would be to not add the chunk if it does not contain any choice (line 1094)

Reply:
updated code to test before adding chunk